server-common/src/main/java/org/apache/kafka/server/common/TopicIdPartition.java

```diff
@@ -14,43 +14,62 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+package org.apache.kafka.server.common;
 
-package org.apache.kafka.controller;
-
-import java.util.Objects;
 import org.apache.kafka.common.Uuid;
 
-final class TopicIdPartition {
+/**
+ * Represents a partition using its unique topic Id and partition number.
+ */
+public final class TopicIdPartition {
     private final Uuid topicId;
-    private final int partitionId;
+    private final Integer partitionId;
 
-    TopicIdPartition(Uuid topicId, int partitionId) {
+    public TopicIdPartition(Uuid topicId, int partitionId) {
         this.topicId = topicId;
         this.partitionId = partitionId;
     }
 
+    /**
+     * @return Universally unique Id representing this topic partition.
+     */
     public Uuid topicId() {
         return topicId;
     }
 
+    /**
+     * @return The partition Id.
+     */
     public int partitionId() {
         return partitionId;
     }
 
     @Override
     public boolean equals(Object o) {
-        if (!(o instanceof TopicIdPartition)) return false;
-        TopicIdPartition other = (TopicIdPartition) o;
-        return other.topicId.equals(topicId) && other.partitionId == partitionId;
+        if (this == o) {
```

Comment:
nit: Should we keep the previous implementation for equals, hashCode and toString? We just want to move the class so we should not change it.

Reply:
I figured while we're at it we could update the methods according to when we reviewed TopicIdPartition file that I created in the coordinator common class. I could revert it if you feel the old one is fine